federation/handle.go

```diff
@@ -91,29 +84,22 @@ func MakeRespMakeJoin(s *Server, room *ServerRoom, userID string) (resp fclient.
 // or dealing with HTTP responses itself.
 func MakeRespMakeKnock(s *Server, room *ServerRoom, userID string) (resp fclient.RespMakeKnock, err error) {
 	// Generate a knock event
-	proto := gomatrixserverlib.ProtoEvent{
-		SenderID:   userID,
-		RoomID:     room.RoomID,
-		Type:       "m.room.member",
-		StateKey:   &userID,
-		PrevEvents: []string{room.Timeline[len(room.Timeline)-1].EventID()},
-		Depth:      room.Timeline[len(room.Timeline)-1].Depth() + 1,
-	}
-	err = proto.SetContent(map[string]interface{}{"membership": spec.Join})
-	if err != nil {
-		err = fmt.Errorf("make_knock cannot set membership content: %w", err)
-		return
-	}
-	stateNeeded, err := gomatrixserverlib.StateNeededForProtoEvent(&proto)
+	proto, err := room.ProtoEventCreator(Event{
+		Type:     "m.room.member",
+		StateKey: &userID,
+		Content: map[string]interface{}{
+			"membership": spec.Join, // XXX this feels wrong?
```

Comment:
Why? It seems okay to me.

Reply:
It's `MakeRespMakeKnock` so surely should be knock not join.